### PR TITLE
Fixed some memory and loading issues in meshes

### PR DESCRIPTION
**Pull request by Verkhovskaya**

### CODE REVIEW

File: src/Sector.ts [hunk 1]
```diff
@@ -49,6 +49,19 @@ export default class Sector {
     }
   }
 
+  *traverseSectorsBreadthFirst(): IterableIterator<Sector> {
+    const stack: Sector[] = [this];
+    while (stack.length > 0) {
+      const nextSector = stack.shift();
+      nextSector!.children.forEach(child => {
+        if (child !== undefined) {
```
Comment (anderscognite) [inline src/Sector.ts]: Can the child be `undefined`?

Comment (anderscognite) [reply src/Sector.ts]: If so, should we rather do `child != null`?

File: src/customFileParser/unpackGeometry/main.ts [hunk 2]
```diff
@@ -38,23 +39,23 @@ function findOrCreateDestinationGroup(
...
     return destinationGroup;
   } else {
-    if (originalSector.children.length === 0) {
-      const capacity = renderedPrimitiveInfo.count * numOfGeometries;
-      return createGroup(originalSector, renderedPrimitiveInfo.name, capacity);
-    } else {
-      return createGroup(originalSector, renderedPrimitiveInfo.name, 1000);
-    }
+    // @ts-ignore
+    const capacity = Math.min(window.maxCapacity,
```
Comment (anderscognite) [inline src/customFileParser/unpackGeometry/main.ts]: `window.` should probably not be used :D

File: src/customFileParser/unpackGeometry/main.ts [hunk 3]
```diff
@@ -63,12 +64,14 @@ function updateDestinationGroups(
   currentSector: Sector,
   primitiveCompressedData: any,
   numOfGeometries: number,
+  groupSpaceRequirementsPerSector: any,
```
Comment (anderscognite) [inline src/customFileParser/unpackGeometry/main.ts]: Why is this of type `any` (everywhere)?

File: src/customFileParser/unpackGeometry/main.ts [hunk 4]
```diff
@@ -78,15 +81,17 @@ function unpackFilePrimitive(
   primitiveCompressedData: CompressedGeometryData,
   uncompressedValues: any,
   treeIndexNodeIdMap: any,
-  colorMap: any) {
+  colorMap: any,
```
Comment (anderscognite) [inline src/customFileParser/unpackGeometry/main.ts]: Why is this any? We have a `ColorMap` type.

File: src/customFileParser/unpackGeometry/main.ts [hunk 5]
```diff
@@ -102,10 +107,46 @@ export function unpackPrimitives(
   treeIndexNodeIdMap: number[],
   colorMap: THREE.Color[]) {
 
-  for (const sector of rootSector!.traverseSectors()) {
+  const groupSpaceRequirementsPerSector: any = {};
```
Comment (anderscognite) [inline src/customFileParser/unpackGeometry/main.ts]: Create a type for this.

File: src/customFileParser/unpackGeometry/main.ts [hunk 6]
```diff
@@ -102,10 +107,46 @@ export function unpackPrimitives(
...
+  for (const sector of rootSector.traverseSectors()) {
+    const renderedPrimitiveCount: any = {};
```
Comment (anderscognite) [inline src/customFileParser/unpackGeometry/main.ts]: I don't understand the names here. You have both `renderedPrimitiveCount` as a map, but also `renderedPrimitive.count`.. What is `filePrimitive.count`? I think some renaming / comments would be good.

**Review by anderscognite — CHANGES_REQUESTED** [review]
Looking good, but I'd love to see these things fixed / commented :)

File: src/customFileParser/unpackGeometry/main.ts (hunk 2 above)
Comment (Verkhovskaya) [reply src/customFileParser/unpackGeometry/main.ts]: Whoops, that got into the wrong branch. (I was using window extensively for my performance tests)

File: src/Sector.ts (hunk 1 above)
Comment (Verkhovskaya) [reply src/Sector.ts]: Oops, that's a left-over from debugging.

File: src/customFileParser/unpackGeometry/main.ts (hunk 3 above)
Comment (Verkhovskaya) [reply src/customFileParser/unpackGeometry/main.ts]: Fixed

File: src/customFileParser/unpackGeometry/main.ts (hunk 4 above)
Comment (Verkhovskaya) [reply src/customFileParser/unpackGeometry/main.ts]: Fixed

File: src/customFileParser/unpackGeometry/main.ts (hunk 5 above)
Comment (Verkhovskaya) [reply src/customFileParser/unpackGeometry/main.ts]: Fixed

File: src/customFileParser/unpackGeometry/main.ts (hunk 6 above)
Comment (Verkhovskaya) [reply src/customFileParser/unpackGeometry/main.ts]: Fixed

**Review by anderscognite — APPROVED** [review]

**Pull request merged**